darwin/Platform.c

```diff
@@ -120,21 +120,65 @@ const MeterClass* const Platform_meterTypes[] = {
    NULL
 };
 
-static double Platform_nanosecondsPerMachTick = 1.0;
+static void Platform_getCPUBrandString(char *cpu_brand_string, size_t cpu_brand_string_size) {
+   if (sysctlbyname("machdep.cpu.brand_string", cpu_brand_string, &cpu_brand_string_size, NULL, 0) == -1) {
+      CRT_fatalError("Could not determine what CPU we are running on");
+   }
+}
 
-static double Platform_schedulerTicksPerNS = -1;
+// Adapted from https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment
+static bool Platform_isRunningUnderRosetta2() {
+   int ret = 0;
+   size_t size = sizeof(ret);
+   if (sysctlbyname("sysctl.proc_translated", &ret, &size, NULL, 0) == -1) {
+      if (errno == ENOENT) return 0;
+      CRT_fatalError("Could not determine if this process was running in a translation environment like Rosetta 2.");
+   }
+   return ret;
+}
 
-void Platform_init(void) {
+static double Platform_calculateNanosecondsPerMachTick() {
    // Check if we can determine the timebase used on this system.
    // If the API is unavailable assume we get our timebase in nanoseconds.
-#ifdef HAVE_MACH_TIMEBASE_INFO
-   mach_timebase_info_data_t info;
-   mach_timebase_info(&info);
-   Platform_nanosecondsPerMachTick = (double)info.numer / (double)info.denom;
-#else
-   Platform_nanosecondsPerMachTick = 1.0;
+#ifndef HAVE_MACH_TIMEBASE_INFO
+   return 1.0;
 #endif
 
+   mach_timebase_info_data_t info;
+
+   /* WORKAROUND for `mach_timebase_info` giving incorrect values on M1 under Rosetta 2.
+    *    rdar://FB9546856 https://openradar.appspot.com/radar?id=5055988478509056
+    *
+    *    We don't know exactly what feature/attribute of the M1 chip causes this mistake under Rosetta 2.
+    *    Until we have more Apple ARM chips to compare against, the best we can do is special-case
+    *    the "Apple M1" chip specifically when running under Rosetta 2.
+    */
+
+   size_t cpu_brand_string_size = 1024;
```

Comment:
Can we start to bring darwin back into the rest of the htop fold and use camel case variables?

Reply:
IDK, you tell me! Should I?

Reply:
Yes please.  For now just in the areas you're modifying, over time let's move towards consistency throughout htop.

Reply:
Done.